### ISSUE TITLE
Not working with VS Code

### ISSUE THREAD

**Issue by googoling** [eczarny/spectacle#903]
It's not working well with the VS code. Please add support for VS code.

**Comment by kainzjoh** [comment]
@googoling I just tried with `spectacle v1.2 (672)` and `vscode Version (1.32.3)` and could not see any issues with window resizing or moving around. could you please specify?

**Comment by dfsdevops** [comment]
I am also having this issue.  Spectacle 1.2 and vscode Version 1.33.0

I have checked to make sure it's not disabled in the preferences. I didn't have this problem until a few weeks ago and I haven't been able to get it working since.

**Comment by jhoch** [comment]
Bizarrely, I've just started having this issue as of this afternoon, with the VSCode 1.34.0 update.  Also using Spectacle 1.2.

Checked that Spectacle wasn't disabled.  Spectacle is capturing the key combinations; when I go to the Keyboard shortcuts in VSCode and use key capture, alt+cmd+F doesn't show up, although alt+cmd+A does.

**Comment by originalix** [comment]
Before VSCode 1.35.1 update, spectacle 1.2 worked well. But now i update VSCode to 1.35.1, spectacle not work. Please help me. thank you.

**Comment by ggedde** [comment]
Same here vs code 1.35.1 and Spectacle 1.2
I changed my keybindings to something else that VS Code captures, but still nothing.

**Comment by ggedde** [comment]
Ok, well I installed the VS Code Insider build and Spectacle worked.
So then I went back to my other VS Code and disabled all Extensions and restarted and now Spectacle works.

So then I re-enabled all extensions and Spectacle still works.

So not sure why, but disabling Extensions and Restarting then Re-Enabling Extensions and Restarting seems to have worked for me.

**Comment by originalix** [comment]
> Ok, well I installed the VS Code Insider build and Spectacle worked.
> So then I went back to my other VS Code and disabled all Extensions and restarted and now Spectacle works.
> 
> So then I re-enabled all extensions and Spectacle still works.
> 
> So not sure why, but disabling Extensions and Restarting then Re-Enabling Extensions and Restarting seems to have worked for me.

I used version of 1.35.0 back.

**Comment by adamxtokyo** [comment]
I'm also having this problem and I'm not willing to use the beta version of VS Code to possibly get it working...

The main issue is I can't resize VS Code with keyboard shortcuts. I removed conflicting keyboard shortcuts in VS Code (Code > Preferences > Keyboard Shortcuts [cmd+K, cmd+S]), but still no luck. I tried moving it to the left/right half of the screen without any success.

I'm using Spectacle 1.2 (672), and here's my VS Code version info:
Version: 1.38.1
Commit: b37e54c98e1a74ba89e03073e5a3761284e3ffb0
Date: 2019-09-11T13:31:32.854Z
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Darwin x64 18.7.0

**Comment by rxhanson** [comment]
I recently rewrote Spectacle in Swift as a new app called [Rectangle](https://rectangleapp.com), and from what I can tell VS Code works just fine with Rectangle. Feel free to try it out. If VSCode doesn't work, I can look at it.

Note that Spectacle hasn't been maintained/updated in over 2 years. I believe the problem it has with VSCode is rooted in the shortcut manager built into Spectacle, and Rectangle uses MASShortcut instead.

**Comment by ggedde** [comment]
@adamxtokyo,  you don’t need to use the beta version. That’s what I did at first but later on I went back to the stable version but all I had to do was disable all extensions restart VS code enable all extensions  restart VS code again and then it worked. Have you tried that yet?

**Comment by adamxtokyo** [comment]
@ggedde thanks for the tip, but I ended up going for Rectangle instead 😁

@rxhanson awesome work! 👏 I'm gonna recommend this to all my work colleagues as Spectacle is very popular here. Thanks for creating Rectangle and keep up the good work!

**Comment by jakehopking** [comment]
I found this today after VS Code updated to 1.38.1 and Spectacle ceased to function.
I've tried the rather nice looking Rectangle app by @rxhanson, however VS Code still won't respond to any shortcuts specified?? Any ideas?

**Comment by ggedde** [comment]
@jakehopking,
Have you tried disabling all extensions restart VS code enable all extensions restart VS code again?

**Comment by rxhanson** [comment]
@jakehopking, I can take a look at what's going on with Rectangle if you file a bug there (https://github.com/rxhanson/Rectangle/issues) with more details on what you mean by VS Code not responding to shortcuts specified. As in, post if there are specific shortcuts you see issues with, or if it is when you try to record a shortcut in VSCode that is already taken by Rectangle.

**Comment by borekb** [comment]
In my case, keyboard shortcuts don't work but clicking Spectacle icon and selecting e.g. 'Right Half' works. So it's probably related to keyboard shortcuts, as indicated by @rxhanson in https://github.com/eczarny/spectacle/issues/903#issuecomment-531758920.

BTW I tried disabling all my extensions and restarting VSCode. It didn't help.

**Comment by creimers** [comment]
For me it works when all VSCode extensions are disabled. Once they are enabled again, it does not work. Looks like there must be some kind of keyboard shortcuts conflict with one of the extensions?

**Comment by James-E-Adams** [comment]
In case people are still curious if this is still relevant, it worked for me. 

> Ok, well I installed the VS Code Insider build and Spectacle worked.
> So then I went back to my other VS Code and disabled all Extensions and restarted and now Spectacle works.
> 
> So then I re-enabled all extensions and Spectacle still works.
> 
> So not sure why, but disabling Extensions and Restarting then Re-Enabling Extensions and Restarting seems to have worked for me.

**Comment by g-patel** [comment]
unfortunately, I stared having this with vscode 1.43.2 (and spectacle 1.2) It was all fine all along before.

**Comment by g-patel** [comment]
well, after quitting and restating vscode, it started working again. I think the issue might be on the vscode front. not spectacle to blame here.!

**Comment by sabrina-s** [comment]
Anyone else having this issue still with vscode 1.50.1 and Spectacle 1.2? 😞
I've checked to make sure I don't have any keymaps on my vscode that match the ones used on Spectacle.

**Comment by kmcgrevey** [comment]
Yeah -- issues here with `VSCode 1.50.1` and `Spectacle 1.2` as well.

Conflict with tabbing between split panes in VSCode.  No matter what shortcuts I set/try in VSCode to not duplicate what is in Spectacle it simply does not tab.  When I disable Spectacle (for VSCode), the VSCode shortcut (option + cmd + right-arrow) work flawlessly.  SUPER annoying.

The disable/enable extensions experiment did not work for me.

May try `Rectangle` mentioned above.  Anybody have luck with it?

**Comment by rxhanson** [comment]
@kmcgrevey I've never personally had any issues with Rectangle and VS Code, but some users have reported having issues that were resolved by restarting VS Code (https://github.com/rxhanson/Rectangle/issues/43). Perhaps restarting VS Code can help in Spectacle as well.

There's not really anything that can be done from Rectangle's perspective on it, so my recommendation would be to file an issue with VS Code if you continue to experience it in Rectangle.

**Comment by kmcgrevey** [comment]
Oh hell yeah **@rxhanson** !  Just fired up `Rectangle` . . . slam dunk!  Gotta get familiar with the shortcuts, but they seem really intuitive and out of the way of VSCode.  Probly just learn 'em and run with it as-is.  Nice job man!

**Comment by yuanhengzhao** [comment]
I am on mac. And reboot the system works for me.

**Comment by ManniCS** [comment]
Restarting VSCode also worked for me.
